hal/inc/ble_hal.h

```diff
@@ -100,6 +100,12 @@ typedef enum hal_ble_scan_fp_t {
                                                                    address is a resolvable private address that cannot be resolved. */
 } hal_ble_scan_fp_t;
 
+typedef enum hal_ble_phys_t {
+    BLE_PHYS_1MBPS             = 0,  /**< Standard 1MBPS format compatible with all BLE 4/5 devices */
+    BLE_PHYS_CODED             = 1,  /**< Coded Phy, 125 kbps, gives longer-range; BLE 5 only */
+    BLE_PHYS_BOTH              = 2   /**< Scanning only: simultaneous scan for both 1MBPS and CODED formats */
```

Comment:
Sorry for incorrectly suggested to not use bit masks in previous comment. After seeking more about BLE PHYs, It appears that a BLE 5 enabled chip may also support 2MBPS PHY. Maybe in the future standard more PHYs will be supported. If not using bit masks, to support multiple PHYs at the same time we have to introduce more enum values like `BLE_PHYS_1MBPS_CODED_XXX`, `BLE_PHYS_1MBPS_CODED_YYY`, `BLE_PHYS_1MBPS_CODED_XXX_YYY` etc. That's a bad design. So let's still use bit masks and map the Nordic defined values for now. So my suggestion would be like this:
```
typedef enum hal_ble_phys_t {
    BLE_PHYS_AUTO = 0x00,
    BLE_PHYS_1MBPS = 0x01,
    ...
} hal_ble_phys_t;
```

Reply:
I have reverted to the Nordic enum values as requested.  Also updated the test app.  When combining the enums with a bitwise OR, it was necessary to cast the result back to hal_ble_phy_t, as you will see in the test app. I have a 2nd Boron on order and will be working on a BLE 5 Coded Phy Advertising update for an end-to-end test.